importing-esi-nodes.md

```diff
@@ -0,0 +1,170 @@
+
+    ```
+    openstack network create mynetwork
+    ```
+
+2. Create a subnet. A subnet defines a pool of addresses that can be assigned to nodes on the network. The `--subnet-range` option sets the CIDR for the network, while the `--allocation-pool` option restricts the range of automatically assigned addresses (which is useful if you will need to manually allocate some addresses for vips, etc):
+
+    ```
+    openstack subnet create mynetwork-subnet --network mynetwork \
+      --subnet-range 10.10.10.0/24 \
+      --allocation-pool start=10.10.10.10,end=10.10.10.254
+    ```
+
+### Attach the network to the router
+
+1. Attach the subnet you just created to your router:
+
+    ```
+    openstack router add subnet mynetwork-subnet
+    ```
+
+## Register nodes with ACM
+
+In order to register the nodes with ACM, we need to boot them using the discovery image provided by our target infrastructure environment. 
+
+1. Get the URL for the discovery image:
+
+    ```
+    discovery_url=$(kubectl get infraenv myinfraenv -o jsonpath='{ .status.isoDownloadURL }')
+    ```
+
+2. Remove any existing network attachments from your nodes. This isn't necessary if you have just acquired the nodes; you only need to do this if the nodes were previously attached to a network:
+
+    ```
+    openstack esi node network detach --all mynode
+    ```
+
+3. Attach the node to the ESI provisioning network:
+
+    ```
+    openstack esi node network attach --network provisioning mynode
+    ```
+
+4. Configure the node to boot from the discovery image:
+
+    ```
+    openstack baremetal node set mynode \
+      --instance-info deploy_interface=ramdisk \
+      --instance-info boot_iso="$discovery_url"
+    ```
+
+5. Boot the node:
+
+    ```
+    openstack baremetal node deploy mynode
+    ```
+
+Wait for the nodes to boot and appear as agents in the infrastructure environment.
+
+## Attach the nodes to your target subnet
+
+Once the nodes have registered with the infrastructure environment, you need to move them off the provisioning network and onto the network we created earlier:
+
+1. Detach the node from the provisioning network:
+
+    ```
+    openstack esi node network detach --all mynode
+    ```
+
+1. Attach the node your target network:
+
+    ```
+    openstack esi node network attach --network mynetwork mynode
+    ```
+
+You will see that the discovered ip addresses for the node are updated to reflect the new network attachment.
+
+## Update agent metadata
+
+When a node is registered as an agent in ACM, it is identified by a UUID. There is a display hostname associate with the agent; in the absence of reverse dns, the hostname will be generated from the ip address of the node at the time it was registered. Since we have moved the node onto a different network from the one on which it was discovered, these hostnames have no obvious relation to the node. We would like the agent hostnames to reflect the ESI nodename of the associated node.
```

Comment:
"associated"; also DNS and IP would be capitalized. I wouldn't mention any of this if I didn't have a general question :)